hawc/apps/riskofbias/actions/rob_approach.py

```diff
@@ -0,0 +1,105 @@
+import json
+from pathlib import Path
+from textwrap import dedent
+from typing import Any, Dict, List
+
+from django.conf import settings
+from django.db import transaction
+from django.db.models import IntegerChoices
+
+from ...assessment.models import Assessment, Log
+from ..models import RiskOfBias, RiskOfBiasAssessment, RiskOfBiasDomain, RiskOfBiasMetric
+
+
+class RobApproach(IntegerChoices):
+    NTP_OHAT = 1, "NTP OHAT"
+    EPA_IRIS = 2, "USEPA IRIS"
+
+    def load_fixture_data(self) -> Dict[str, Any]:
+        """Return fixture data as generated from data files."""
+        _fixture = {
+            self.NTP_OHAT: "ohat_study_quality_defaults.json",
+            self.EPA_IRIS: "iris_study_quality_defaults.json",
+        }
+        p = Path(settings.PROJECT_PATH) / "apps/riskofbias/fixtures" / _fixture[self.value]
+        return json.loads(p.read_text())
+
+
+@transaction.atomic
+def load_approach(assessment_id: int, approach: RobApproach, user_id: int = None):
+    """
+    Construct default risk of bias domains/metrics for an assessment.
+    """
+    # fetch data
+    data = approach.load_fixture_data()
+
+    # log about changes being made
+    domains = RiskOfBiasDomain.objects.filter(assessment_id=assessment_id)
+    robs = RiskOfBias.objects.filter(study__assessment=assessment_id)
+    log_message = dedent(
+        f"""\
+        Loading Risk of Bias approach {approach.name} into assessment {assessment_id}
+        Deleting {domains.count()} RiskOfBiasDomain objects
+        Deleting {robs.count()} RiskOfBias objects"""
+    )
+    Log.objects.create(assessment_id=assessment_id, user_id=user_id, message=log_message)
+
+    # delete existing data (recursively deletes domains, metrics, scores, etc)
+    domains.delete()
+    robs.delete()
+
+    # load help text
+    settings = RiskOfBiasAssessment.objects.get(assessment_id=assessment_id)
+    settings.help_text = data["help_text"]
+    settings.save()
+
+    # create domains and metrics
+    metrics: List[RiskOfBiasMetric] = []
+    for sort_order, domain_data in enumerate(data["domains"], start=1):
+        metrics_data = domain_data.pop("metrics")
+        domain = RiskOfBiasDomain.objects.create(assessment_id=assessment_id, **domain_data)
+        metrics.extend(
+            [
+                RiskOfBiasMetric(domain_id=domain.id, **metric_data)
```

Comment:
`sort_order` will need to be passed to the constructors here and elsewhere once the changes from `main` are merged into this branch.

Reply:
now added since we merged the other PR in...